drivers/platform/stm32/stm32_tdm.c

```diff
@@ -219,3 +228,46 @@ int32_t stm32_tdm_read(struct no_os_tdm_desc *desc,
 
 	return ret;
 }
+
+/**
+ * @brief Stop SAI DMA transfer
+ * @param desc - The TDM descriptor.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t stm32_stop_tdm_dma_transfer(struct no_os_tdm_desc *desc)
+{
+	int32_t	ret;
+	struct stm32_tdm_desc *tdesc;
+
+	if (!desc)
+		return -EINVAL;
+
```

Comment:
I think an extra check is needed here:

```
if (blocking)
    return -ENOSYS;
```

Reply:
it is up to you how you detect if user configured the API in blocking or non-blocking

Reply:
(and this comment applies to the implementations of pause and resume too)

Reply:
The blocking check has been made with respect to the definition of rx_callback.